libnetdata/avl/avl.h

```diff
@@ -34,13 +34,13 @@ typedef struct avl {
 } avl;
 
 /* An AVL tree */
-typedef struct avl_tree {
+typedef struct avl_tree_type {
     avl *root;
     int (*compar)(void *a, void *b);
-} avl_tree;
+} avl_tree_type;
 
 typedef struct avl_tree_lock {
-    avl_tree avl_tree;
+    avl_tree_type avl_tree;
```

Comment:
```suggestion
    avl_tree avl_tree_instance;
```
I don't like how changes propagated outside of libnetdata. Let's go the other way around, similar to what you did with `struct avl`.

Reply:
I originally tried that as well. But the avl_tree member is being accessed on many places, so it propagates outside even more. Moreover, as part of the app config structure (appconfig.h, struct config) it is being used and accessed often by collector plugins, so changing the member name has bigger potential to break third-party plugins than the type name change.
But both variants are possible, so if you think changing member name is better, I can do that.

Reply:
No, you are right, there are a bunch of initializers that touch the `avl_tree` field at various places. It's probably not worth it.